src/schema/rules/dataset_metadata.yaml

```
@@ -0,0 +1,53 @@
+dataset_description:
+    selectors:
+    - path == /dataset_description.json
+    fields:
+        Name: REQUIRED
+        BIDSVersion: REQUIRED
+        DatasetType: RECOMMENDED
+        License: RECOMMENDED
+        Authors: OPTIONAL
+        Acknowledgements: OPTIONAL
+        HowToAcknowledge: OPTIONAL
+        Funding: OPTIONAL
+        EthicsApprovals: OPTIONAL
+        ReferencesAndLinks: OPTIONAL
+        DatasetDOI: OPTIONAL
+        GeneratedBy: RECOMMENDED
+        GeneratedBy[].Name: REQUIRED
+        GeneratedBy[].Version: RECOMMENDED
+        GeneratedBy[].Description: OPTIONAL
+        GeneratedBy[].CodeURL: OPTIONAL
+        GeneratedBy[].Container: OPTIONAL
+        GeneratedBy[].Container.Type: RECOMMENDED
+        GeneratedBy[].Container.Tag: RECOMMENDED
+        GeneratedBy[].Container.URI: RECOMMENDED
```

For an object, would the convention then be `GeneratedBy.Name` or `GeneratedBy{}.Name` (in case we want to allow metadata keys with periods in them without assuming they're dictionaries)?

---

I assumed that `.` would do the job, but `GeneratedBy{}.Name` would be more explicit. Happy to have more opinions here.

---

To make your suggestion explicit, I think

```suggestion
        GeneratedBy[].Name: REQUIRED
        GeneratedBy[].Version: RECOMMENDED
        GeneratedBy[].Description: OPTIONAL
        GeneratedBy[].CodeURL: OPTIONAL
        GeneratedBy[].Container: OPTIONAL
        GeneratedBy[].Container{}.Type: RECOMMENDED
        GeneratedBy[].Container{}.Tag: RECOMMENDED
        GeneratedBy[].Container{}.URI: RECOMMENDED
```

---

That looks good to me.